ethereum_history_api/circuits/lib/src/verifiers/split_into_tx_type_and_rlp.nr

```diff
@@ -0,0 +1,16 @@
+use crate::misc::arrays::memcpy;
+use crate::receipt::TxType;
+use crate::verifiers::receipt::{MAX_ENCODED_RECEIPT_LENGTH, MAX_RECEIPT_RLP_LENGTH};
+
+pub(crate) fn split_into_tx_type_and_rlp<MAX_ENCODED_LENGTH, MAX_RLP_LENGTH>(tx_type: TxType, encoded: [u8; MAX_ENCODED_LENGTH]) -> (u8, [u8; MAX_RLP_LENGTH]) {
```

Comment:
If we only use tx_type to check if it's a legacy receipt/transaction then maybe change tx_type parameter to is_legacy: bool parameter and return:
```rust
let legacy_type = 0
(legacy_type, rlp)
```

It's a suggestion, not sure it that's a better way

Reply:
I feel like if we pass tx_type as parameter it can be confusing that we don't check inside the function that actual_tx_type == tx_type

Reply:
Nice catch. I was under the impression that I've already fixed it somewhere, but it probably got lost. Thanks